packages/core/src/repositories/connectedEvaluationsRepository.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  and,
+  count,
+  eq,
+  getTableColumns,
+  isNotNull,
+  max,
+  sql,
+  sum,
+} from 'drizzle-orm'
+
+import { ConnectedEvaluation, DocumentVersion } from '../browser'
+import { LatitudeError, Result, TypedResult } from '../lib'
+import {
+  connectedEvaluations,
+  documentLogs,
+  evaluationResults,
+  evaluations,
+  providerLogs,
+} from '../schema'
+import { DocumentVersionsRepository } from './documentVersionsRepository'
+import Repository from './repository'
+
+const tt = getTableColumns(connectedEvaluations)
+
+export type ConnectedDocumentWithMetadata = DocumentVersion & {
+  projectId: number
+  evaluationLogs: number
+  totalTokens: number
+  costInMillicents: number
+}
+
+export class ConnectedEvaluationsRepository extends Repository<
+  typeof tt,
+  ConnectedEvaluation
+> {
+  get scope() {
+    return this.db
+      .select(tt)
+      .from(connectedEvaluations)
+      .innerJoin(
+        evaluations,
+        eq(connectedEvaluations.evaluationId, evaluations.id),
+      )
+      .where(eq(evaluations.workspaceId, this.workspaceId))
+      .as('connectedEvaluationsScope')
+  }
+
+  async findByEvaluationId(id: number) {
+    const result = await this.db
+      .select()
+      .from(this.scope)
+      .where(eq(this.scope.evaluationId, id))
+
+    return Result.ok(result)
+  }
+
+  async findByDocumentUuid(uuid: string) {
+    const result = await this.db
+      .select()
+      .from(this.scope)
+      .where(eq(this.scope.documentUuid, uuid))
+
+    return Result.ok(result[0]!)
+  }
+
+  async getConnectedDocumentsWithMetadata(
+    evaluationId: number,
+  ): Promise<TypedResult<ConnectedDocumentWithMetadata[], LatitudeError>> {
```

Comment:
Attention: black magic ahead. Complex sql query to fetch aggregated data from connected evaluations. The idea being: Display all documents connected to this evaluation, and the costs and metadata of the results. The problem is that the relations from connectedEvaluations to this info is not really straight-forward.